modules/registry-refactoring/src/main/java/org/apache/airavata/registry/core/repositories/appcatalog/UserResourceProfileRepository.java

```diff
@@ -0,0 +1,227 @@
+
+        if(!userStoragePreferenceList.isEmpty() && userStoragePreferenceList != null) {
+            logger.debug("Return the record (there is only one record)");
+            return userStoragePreferenceList.get(0);
+        }
+
+        return null;
+    }
+
+    @Override
+    public List<UserResourceProfile> getAllUserResourceProfiles() throws AppCatalogException {
+        List<UserResourceProfile> userResourceProfileList = select(QueryConstants.GET_ALL_USER_RESOURCE_PROFILE, 0);
+        return userResourceProfileList;
+    }
+
+    @Override
+    public List<UserComputeResourcePreference> getAllUserComputeResourcePreferences(String userId, String gatewayId) throws AppCatalogException {
+        UserComputeResourcePreferenceRepository userComputeResourcePreferenceRepository = new UserComputeResourcePreferenceRepository();
+        Map<String, Object> queryParameters = new HashMap<>();
+        queryParameters.put(DBConstants.UserComputeResourcePreference.USER_ID, userId);
+        queryParameters.put(DBConstants.UserComputeResourcePreference.GATEWAY_ID, gatewayId);
+        List<UserComputeResourcePreference> userComputeResourcePreferenceList =
+                userComputeResourcePreferenceRepository.select(QueryConstants.GET_ALL_USER_COMPUTE_RESOURCE_PREFERENCE, -1, 0, queryParameters);
+        return userComputeResourcePreferenceList;
+    }
+
+    @Override
+    public List<UserStoragePreference> getAllUserStoragePreferences(String userId, String gatewayId) throws AppCatalogException {
+        UserStoragePreferenceRepository userStoragePreferenceRepository = new UserStoragePreferenceRepository();
+        Map<String, Object> queryParameters = new HashMap<>();
+        queryParameters.put(DBConstants.UserStoragePreference.USER_ID, userId);
+        queryParameters.put(DBConstants.UserStoragePreference.GATEWAY_ID, gatewayId);
+        List<UserStoragePreference> userStoragePreferenceList =
+                userStoragePreferenceRepository.select(QueryConstants.GET_ALL_USER_STORAGE_PREFERENCE, -1, 0, queryParameters);
+        return userStoragePreferenceList;
+    }
+
+    @Override
+    public List<String> getGatewayProfileIds(String gatewayName) throws AppCatalogException {
+        Map<String, Object> queryParameters = new HashMap<>();
+        queryParameters.put(DBConstants.UserResourceProfile.GATEWAY_ID, gatewayName);
+        List<UserResourceProfile> userResourceProfileList = select(QueryConstants.GET_ALL_GATEWAY_ID, -1, 0, queryParameters);
+        List<String> gatewayIdList = new ArrayList<>();
+        for (UserResourceProfile userResourceProfile : userResourceProfileList) {
+            gatewayIdList.add(userResourceProfile.getGatewayID());
+        }
+        return gatewayIdList;
+    }
+
+    @Override
+    public String getUserNamefromID(String userId, String gatewayID) throws AppCatalogException {
```

Comment:
Hmm, what is this used for, this seems weird?

Reply:
I don't really know. This method is a part of the UsrResourceProfile interface. So, I had to override it. But it doesn't make sense at all.

Reply:
Okay, that's fine, let's keep it for now since it is part of the interface.

Reply:
Sure.